src/smt/theory_str_noodler/ca_str_constr.h

```diff
@@ -0,0 +1,170 @@
+            create_aut_matrix(diseq, aut_ass);
+        }
+
+        /**
+         * @brief Get state in unified automaton (where all automata in matrix are unioned).
+         * 
+         * @param copy Index of the copy
+         * @param var Index of the variable (index in @p var_order)
+         * @param state State of the particular automaton at [ @p copy, @p var ]
+         * @return mata::nfa::State State in the big NFA
+         */
+        mata::nfa::State get_union_state(size_t copy, size_t var, mata::nfa::State state) const {
+            return get_offset(copy, var) + state;
+        }
+
+        /**
+         * @brief Unify all particular automata into a single NFA.
+         * 
+         * @return mata::nfa::Nfa Big NFA
+         */
+        mata::nfa::Nfa union_matrix() const;
+
+        const std::vector<BasicTerm>& get_var_order() const {
+            return this->var_order;
+        }
+
+        void set_aut(size_t copy, size_t var, const mata::nfa::Nfa& aut, bool re_offset = false) {
+            this->aut_matrix[copy][var] = aut;
+            if(re_offset) {
+                recompute_offset();
+            }
+        }
+
+        const mata::nfa::Nfa& get_aut(size_t copy, size_t var) const {
+            return this->aut_matrix[copy][var];
+        }
+    };
+
+    /**
+     * @brief Class for Tag aut generation for diseqations.
+     */
+    class TagDiseqGen {
+
+    private:
+        DiseqAutMatrix aut_matrix;
+        AutAssignment aut_ass;
+        Predicate diseq;
+        ca::CounterAlphabet alph {};
+
+    public:
+        TagDiseqGen(const Predicate& diseq, const AutAssignment& aut_ass) : aut_matrix(diseq, aut_ass), 
+            aut_ass(aut_ass), diseq(diseq), alph() { }
+
+    protected:
+        /**
+         * @brief Replace symbols in a particular variable automaton. Replace original symbols with 
+         * the AtomicSymbols of the form <L,x> ...
+         * 
+         * @param copy Copy identifying particular variable automaton
+         * @param var Variable of the automaton
+         */
+        void replace_symbols(char copy, size_t var);
+
+        /**
+         * @brief Add connections between copies. 
+         * 
+         * @param copy_start Starting copy (transitions source)
+         * @param var Variable
+         * @param aut_union Union automaton contains all copies in a single automaton.
+         */
+        void add_connection(char copy_start, size_t var, mata::nfa::Nfa& aut_union);
+
+    public:
+        /**
+         * @brief Construct tagged automaton for a single disequation.
+         * 
+         * @return ca::CA Tagged automaton.
+         */
+        ca::TagAut construct_tag_aut();
+
+        const DiseqAutMatrix& get_aut_matrix() const {
+            return this->aut_matrix;
+        }
+
+    };
+
+    /**
+     * @brief Get LIA formula for a single disequations. The LIA formula describes all length 
```

Comment:
Should it be `single disequation`? Or is this for all disequations? Because the name of the function seems like it is for all disequations, it even takes `Formula`.

Reply:
So-far just a single disequation, but this function is supposed to solve multiple disequations at some point. So I leave the name in plural.

Reply:
The next sentence has also only 'diseqeuation'. I would also put TODO/NOTE or something, that it handles only one disequation (or can it already handle multiple, just it is used with only one?)

Reply:
I added a TODO comment.